### PR TITLE
[memory][sc-fs-storage][sc-dictionary][sc-rocksdb] Add opportunity to search links by substring

### PR DESCRIPTION
**Pull request by NikitaZotov**
* [x] Read PR [documentation](https://github.com/ostis-ai/sc-machine/blob/main/docs/dev/pr.md/)
* [x] Update changelog
* [x] Update documentation

### CODE REVIEW

File: sc-memory/tests/sc-memory/common/sc-link.cpp [hunk 1]
```diff
@@ -214,3 +214,40 @@ TEST_F(ScLinkTest, set_system_idtf)
...
+  EXPECT_TRUE(str == link3.GetAsString());
+
+  EXPECT_FALSE(ctx.FindLinksByContent("content1").empty());
+  EXPECT_FALSE(ctx.FindLinksByContent("continue").empty());
+  EXPECT_FALSE(ctx.FindLinksByContent("contents_25").empty());
+
+  EXPECT_TRUE(ctx.FindLinksByContentSubstring("content1").size() == 1);
+  EXPECT_TRUE(ctx.FindLinksByContentSubstring("cont").size() == 3);
+  EXPECT_TRUE(ctx.FindLinksByContentSubstring("content").size() == 2);
+  EXPECT_TRUE(ctx.FindLinksByContentSubstring("con").size() == 3);
```
Comment (AlexandrZagorskiy) [inline sc-memory/tests/sc-memory/common/sc-link.cpp:250]: There is no test for an empty result

File: sc-memory/tests/sc-memory/containers/sc-dictionary.cpp [hunk 2]
```diff
@@ -66,3 +66,76 @@ TEST_F(ScDictionaryTest, smoke)
...
+
+  list = sc_dictionary_get_by_substr(dictionary, "letters");
+  it = sc_list_iterator(list);
+  while (sc_iterator_next(it))
+  {
+    auto const addr = SC_ADDR_LOCAL_TO_INT(*(sc_addr*)sc_iterator_get(it));
+    EXPECT_TRUE(SC_ADDR_LOCAL_TO_INT(addr4) == addr);
+  }
+  sc_iterator_destroy(it);
+  sc_list_destroy(list);
```
Comment (AlexandrZagorskiy) [inline sc-memory/tests/sc-memory/containers/sc-dictionary.cpp:140]: What should be happened if the result of sc_dictionary_get_by_substr should return nothing?
Is it works well for russian letters? I think such behavior should be tested

**Review by AlexandrZagorskiy — CHANGES_REQUESTED** [review]

File: sc-memory/tests/sc-memory/common/sc-link.cpp (hunk 1 above)
Comment (NikitaZotov) [reply sc-memory/tests/sc-memory/common/sc-link.cpp:250]: Fixed

File: sc-memory/tests/sc-memory/containers/sc-dictionary.cpp (hunk 2 above)
Comment (NikitaZotov) [reply sc-memory/tests/sc-memory/containers/sc-dictionary.cpp:144]: Fixed

File: sc-memory/sc-core/sc-store/sc-fs-storage/sc_rocksdb_fs_storage.cpp [hunk 3]
```diff
@@ -265,6 +265,56 @@ sc_result sc_rocksdb_fs_storage_find(const sc_check_sum * check_sum, sc_addr **
   return SC_RESULT_OK;
 }
 
+sc_result sc_rocksdb_fs_storage_find_by_susbtr(const sc_char * sc_substr, sc_addr ** result, sc_uint32 * result_count)
```
[the remark this answers is not in the document]
Comment (NikitaZotov) [reply sc-memory/sc-core/sc-store/sc-fs-storage/sc_rocksdb_fs_storage.cpp]: Fixed

**Review by AlexandrZagorskiy — APPROVED** [review]

**Review by MikhailSadovsky — APPROVED** [review]

**Pull request merged**